internal/providers/ionoscloud/ionoscloud.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The IONOSCloud provider fetches configurations from the userdata available in
+// the injected file /var/lib/cloud/seed/nocloud/user-data.
+// NOTE: This provider is still EXPERIMENTAL.
+
+package ionoscloud
+
+import (
+	"bytes"
+	"os"
+
+	"github.com/coreos/ignition/v2/config/v3_5_experimental/types"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+
+	"github.com/coreos/vcontext/report"
+)
+
+const (
+	defaultFilename = "/var/lib/cloud/seed/nocloud/user-data"
```

Comment:
Hmm, do you have documentation for this, I am not having any luck finding the default user-data file location for IONOS

I thought cloud-init's default was something like "/var/lib/cloud/instance/user-data"

Reply:
Additionally It might be good to add a reference to the in development metatdata service for this, so we can replace it when completed.

Reply:
Unfortunately, we don't have that documented publicly.
Its just a internal docs. 


However this docs explains how can use Userdata.
https://docs.ionos.com/cloud/compute-services/compute-engine/how-tos/boot-cloud-init


The path is /var/lib/cloud/seed
So it could be detected by the NoCloud Datasource. 
So, im 100% sure of that.

But the /var/lib/cloud/instance is the information about the cloud-init run.

Reply:
@prestist I can back this up as well. We don't have this publicly documented. Unfortunately, as of yet we don't have a metadata server at IONOS and therefore rely on a legacy script that injects data into this file before a VM is booted for the first time.

Reply:
Ah, okay, yeah that makes sense. Ok, would it be okay to just add a comment to point to documentation?

Thank you both for answering my questions.